specifyweb/sp7_models/models.py

```diff
@@ -0,0 +1,102 @@
+    timestampcreated = models.DateTimeField(blank=False, null=False, unique=False, db_column='TimestampCreated', db_index=False, default=timezone.now)
+    timestampmodified = models.DateTimeField(blank=True, null=True, unique=False, db_column='TimestampModified', db_index=False, default=timezone.now)
+    text1 = models.TextField(blank=True, null=True, unique=False, db_column='Text1', db_index=False)
+    text2 = models.TextField(blank=True, null=True, unique=False, db_column='Text2', db_index=False)
+    text3 = models.TextField(blank=True, null=True, unique=False, db_column='Text3', db_index=False)
+
+    # Relationships: Many-to-One
+    collection = models.ForeignKey('specify.Collection', db_column='CollectionID', related_name='collectionobjectgroups', null=False, on_delete=protect_with_blockers)
+    createdbyagent = models.ForeignKey('specify.Agent', db_column='CreatedByAgentID', related_name='+', null=True, on_delete=protect_with_blockers)
+    modifiedbyagent = models.ForeignKey('specify.Agent', db_column='ModifiedByAgentID', related_name='+', null=True, on_delete=protect_with_blockers)
+
+    class Meta:
+        db_table = 'collectionobjectgroup'
+        ordering = ()
+
+    timestamptracker = FieldTracker(fields=['timestampcreated', 'timestampmodified'])
+    save = partialmethod(custom_save)
+
+class CollectionObjectGroupJoin(models.Model): # aka. CoJo or CogJoin
+    specify_model = datamodel.get_table('collectionobjectgroupjoin')
+
+    # ID Field
+    id = models.AutoField(primary_key=True, db_column='collectionobjectgroupjoinid')
+
+    # Fields
+    # TODO: Check if order, version, timestampcreated, and timestampmodified are needed
+    isprimary = models.BooleanField(blank=True, null=True, unique=False, db_column='IsPrimary', db_index=False)
+    issubstrate = models.BooleanField(blank=True, null=True, unique=False, db_column='IsSubstrate', db_index=False)
+    precedence = models.SmallIntegerField(blank=True, null=True, unique=False, db_column='Precedence', db_index=False)
+    order = models.SmallIntegerField(blank=True, null=True, unique=False, db_column='Order', db_index=False)
+    version = models.IntegerField(blank=True, null=True, unique=False, db_column='Version', db_index=False, default=0)
+    timestampcreated = models.DateTimeField(blank=False, null=False, unique=False, db_column='TimestampCreated', db_index=False, default=timezone.now)
+    timestampmodified = models.DateTimeField(blank=True, null=True, unique=False, db_column='TimestampModified', db_index=False, default=timezone.now)
+    text1 = models.TextField(blank=True, null=True, unique=False, db_column='Text1', db_index=False)
+    text2 = models.TextField(blank=True, null=True, unique=False, db_column='Text2', db_index=False)
+    text3 = models.TextField(blank=True, null=True, unique=False, db_column='Text3', db_index=False)
+
+    # Relationships: Many-to-Many
```

Comment:
From schema, this is a bad design.

First, I'm assuming the related_name(s) are incorrect (they seem off)

Say you have two collection objects that belong in one COG. and, this COG happens to a child COG. In your current schema, you'll have two entries in the join table. But, they'll contain duplicated parent-child information (very bad for data normalization). As a solution, can we add the parent cog as a field in CollectionObjectGroup table?

Reply:
Ya, I originally just had `collectionobjectgroupparent` and `collectionobjectchild` as foreign keys.  I thought that was good, but on the whiteboard the design seemed to also have `collectionobjectgroupchild`.  I need to talk some more the geo-analyst team.

Reply:
sounds good, check if having the parent FK on group table will make more sense